lib/components/list-group-item.vue

```diff
@@ -1,79 +1,69 @@
 <template>
-    <component :is="myTag" :class="classObject" ref="item" :to="to" :href="href" :target="target" :rel="rel">
+    <component :is="myTag"
+               :class="classObject"
+               ref="item"
+               v-bind="linkProps">
         <slot></slot>
     </component>
 </template>
 
 <script>
-    import bLink from './link.vue';
+import bLink from './link.vue';
+import { props as originalLinkProps, computed, omitLinkProps } from '../mixins/link'
+// copy link props, but exclude defaults for 'href' & 'to'
+// to ensure proper component tag computation
+const linkProps = Object.assign(omitLinkProps('href', 'to'), {
+    href: { type: originalLinkProps.href.type },
+    to: { type: originalLinkProps.to.type }
+});
 
-    const actionTags = ['a', 'router-link', 'button', 'b-link'];
+const actionTags = ['a', 'router-link', 'button', 'b-link'];
 
-    export default {
-        components: {bLink},
-        computed: {
-            classObject() {
-                return [
-                    'list-group-item',
-                    this.listState,
-                    this.active ? 'active' : null,
-                    this.disabled ? 'disabled' : null,
-                    this.isAction ? 'list-group-item-action' : null
-                ];
-            },
-            isAction() {
-                if (this.action === false) {
-                    return false;
-                }
-                return this.action || this.to || this.href || actionTags.indexOf(this.tag) !== -1;
-            },
-            listState() {
-                return this.variant ? `list-group-item-${this.variant}` : null;
-            },
-            myTag() {
-                if (this.tag) {
-                    return this.tag;
-                }
-                return (this.to || this.href) ? 'b-link' : 'div';
+export default {
+    components: { bLink },
+
+    computed: {
+        linkProps: computed.linkProps,
```

Comment:
I wonder how vue handles caching of this type of computed prop?

I'll take a peak at the source for the `computed.linkProps` to see about reactivity registration.